mods/classic-desktop-icons.wh.cpp

```diff
@@ -105,6 +119,61 @@ HWND FindDesktopWindow(void)
     return NULL;
 }
 
+
+
+using CreateWindowExW_t = decltype(&CreateWindowExW);
+CreateWindowExW_t CreateWindowExW_orig;
+HWND CreateWindowExW_hook(
+    DWORD     dwExStyle,
+    LPCWSTR   lpClassName,
+    LPCWSTR   lpWindowName,
+    DWORD     dwStyle,
+    int       X,
+    int       Y,
+    int       nWidth,
+    int       nHeight,
+    HWND      hWndParent,
+    HMENU     hMenu,
+    HINSTANCE hInstance,
+    LPVOID    lpParam
+)
+{
+    HWND hRes = CreateWindowExW_orig(
+        dwExStyle, lpClassName, lpWindowName,
+        dwStyle, X, Y, nWidth, nHeight, hWndParent,
+        hMenu, hInstance, lpParam
+    );
+
+    /**
+      * Check that the following criteria is met:
+      * - The window has a parent
+      * - lpClassName is non-null and not a bad pointer (MANY windows pass a bad pointer to lpClassName)
+      * - The window's class name is "SysListView32"
+      * - The window's parent's class name is "SHELLDLL_DefView"
+      */
+    if (hWndParent != NULL
+    && lpClassName != NULL
+    && !IsBadStringPtrW(lpClassName, 256))
```

Comment:
Don't use `IsBadXxxPtr`.
https://devblogs.microsoft.com/oldnewthing/20060927-07/?p=29563
See how it's done in other mods that hook `CreateWindowExW` and compare the class name.

For example:
https://github.com/ramensoftware/windhawk-mods/blob/18081443a8c4f967d07c82997d3096f3b9faba6c/mods/taskbar-wheel-cycle.wh.cpp#L1088-L1090

Reply:
`TextualClassName` only makes sense for the argument (i.e. the first usage).